File: ADT/stack-queue-pblm/921.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        size = 0
        open = 0
        for ch in s:
            if ch == '(':
                size += 1
            elif size > 0:
                size -= 1
            else:
                open += 1
        return size + open
        

File: ADT/stack-queue-pblm/921.js
/**
 * @param {string} s
 * @return {number}
 */
var minAddToMakeValid = function (s) {
    let size = 0
    let open = 0
    for (const ch of s) {
        if (ch === '(') {
            size++
        } else if (size > 0) {
            size--
        } else {
            open++
        }
    }
    return size + open
};